build-aux/.run-format.zsh

```diff
@@ -0,0 +1,190 @@
+          log_error "swift-format is not version 508.0.0 or above (found ${swift_format_version})."
+          exit 2
+        fi
+      } else {
+        log_error "No viable swift-format version found (required 508.0.0)"
+        exit 2
+      }
+
+      local -a source_files=((libobs|libobs-*|UI|plugins)/**/*.swift(.N))
+
+      local -a format_args=()
+      ;;
+    *) log_error "Invalid formatter specified: ${1}. Valid options are clang-format, cmake-format, and swift-format."; exit 2 ;;
+  }
+
+  local file
+  local -i num_failures=0
+  if (( check_only )) {
+    for file (${source_files}) {
+      if (( _loglevel > 1 )) log_info "Checking format of ${file}..."
+
+      if ! "${formatter}" ${format_args} "${file}" | diff -q "${file}" - &> /dev/null; then
+        log_error "${file} requires formatting changes."
+
+        if (( fail_on_error == 2 )) return 2;
+        num_failures=$(( num_failures + 1 ))
+      else
+        if (( _loglevel > 1 )) log_status "${file} requires no formatting changes."
+      fi
+    }
+    if (( fail_on_error && num_failures )) return 2;
+  } elif (( ${#source_files} )) {
+    format_args+=(-i)
+    "${formatter}" ${format_args} ${source_files}
+  }
+}
+
+run_format() {
+  if (( ! ${+SCRIPT_HOME} )) typeset -g SCRIPT_HOME=${ZSH_ARGZERO:A:h}
+  if (( ! ${+FORMATTER_NAME} )) typeset -g FORMATTER_NAME=${${(s:-:)ZSH_ARGZERO:t:r}[2]}
+
+  typeset -g host_os=${${(L)$(uname -s)}//darwin/macos}
+  local -i fail_on_error=0
+  local -i check_only=0
+  local -i verbosity=1
+  local -r _version='1.0.0'
+
+  fpath=("${SCRIPT_HOME}/.functions" ${fpath})
+  autoload -Uz set_loglevel log_info log_error log_output log_status log_warning
+
+  local -r _usage="
+Usage: %B${functrace[1]%:*}%b <option>
+
+%BOptions%b:
+
+%F{yellow} Formatting options%f
+ -----------------------------------------------------------------------------
+  %B-c | --check%b                      Check only, no actual formatting takes place
+
+%F{yellow} Output options%f
+ -----------------------------------------------------------------------------
+  %B-v | --verbose%b                    Verbose (more detailed output)
+  %B--fail-[never|error]                Fail script never/on formatting change - default: %B%F{green}never%f%b
+  %B--debug%b                           Debug (very detailed and added output)
+
+%F{yellow} General options%f
+ -----------------------------------------------------------------------------
+  %B-h | --help%b                       Print this usage help
+  %B-V | --version%b                    Print script version information"
+
+  local -a args
+  while (( # )) {
+    case ${1} {
+      --)
+        shift
+        args+=($@)
+        break
+        ;;
+      -c|--check) check_only=1; shift ;;
+      -v|--verbose) (( verbosity += 1 )); shift ;;
+      -h|--help) log_output ${_usage}; exit 0 ;;
+      -V|--version) print -Pr "${_version}"; exit 0 ;;
+      --debug) _verbosity=3; shift ;;
```

Comment:
I may misunderstand the intention. But it should be:
```suggestion
      --debug) verbosity=3; shift ;;
```
No ? Otherwise it's completely unused.

Reply:
Correct, good catch.